src/lib/stack/client.ts

```diff
@@ -48,6 +49,10 @@ export async function stackConnect(client: any, config: StackConnectionConfig) {
     }
     // eslint-disable-next-line new-cap
     const stack = client(clientParams)
+    // Temporary code to set custom host
+    if (REGION_URL_MAPPING[clientParams.region]) {
+      stack.setHost(REGION_URL_MAPPING[clientParams.region])
+    }
```

Comment:
Can we prompt the user to enter a URL, similar to what we did in the marketplace apps, if a specific region/environment URL is not listed? This would provide more flexibility and compatibility for using this tool with any environment. also avoid the rework in the code.

BTW, this suggestion is not directly related to the taxonomy task; it could be considered an enhancement for 'tsgen' because this issue has remained open in all previous versions. If it is feasible, we can implement this change and remove 'dev18' from the URL list.